src/components/LegacyMediaAnalytics/index.js

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+/* eslint-disable import/no-restricted-paths */
+
+import createMediaEventManager from "../MediaCollection/createMediaEventManager";
+import createMediaSessionCacheManager from "../MediaCollection/createMediaSessionCacheManager";
+import createTrackMediaEvent from "../MediaCollection/createTrackMediaEvent";
+import createTrackMediaSession from "../MediaCollection/createTrackMediaSession";
+import createMediaHelper from "./createMediaHelper";
+import {
+  AD_METADATA_KEYS as AdMetadataKeys,
+  AUDIO_METADATA_KEYS as AudioMetadataKeys,
+  EVENT as Event,
+  MEDIA_OBJECT_KEYS as MediaObjectKey,
+  MEDIA_TYPE as MediaType,
+  PLAYER_STATE as PlayerState,
+  STREAM_TYPE as StreamType,
+  VIDEO_METADATA_KEYS as VideoMetadataKeys
+} from "./constants/constants";
+import createGetInstance from "./createGetInstance";
+import { noop, uuid } from "../../utils";
+import createOnBeforeMediaEvent from "../MediaCollection/createOnBeforeMediaEvent";
+
+const createLegacyMediaAnalytics = ({
+  eventManager,
+  sendEdgeNetworkRequest,
+  config,
+  logger,
+  consent
+}) => {
+  const mediaSessionCacheManager = createMediaSessionCacheManager({ config });
+
+  const mediaEventManager = createMediaEventManager({
+    sendEdgeNetworkRequest,
+    config,
+    logger,
+    consent,
+    eventManager
+  });
+
+  const trackMediaEvent = createTrackMediaEvent({
+    mediaSessionCacheManager,
+    mediaEventManager,
+    config
+  });
+  const trackMediaSession = createTrackMediaSession({
+    config,
+    logger,
+    mediaEventManager,
+    mediaSessionCacheManager,
+    legacy: true
+  });
+  const onBeforeMediaEvent = createOnBeforeMediaEvent({
+    mediaSessionCacheManager,
+    logger,
+    config,
+    trackMediaEvent
+  });
+  return {
+    lifecycle: {
+      onBeforeEvent({ mediaOptions, onResponse = noop }) {
+        const { legacy, playerId, getPlayerDetails } = mediaOptions;
```

Comment:
Why do you need to specify legacy? Shouldn't this be a given because you are using LegacyMediaAnalytics component?

Reply:
Because Legacy and Media Collection uses diferent cache instances and I want to save in legacy's cache the sessions that were triggered in legacy mode.